tests/core/test_flow.py

```diff
@@ -1232,7 +1232,16 @@ def test_skip_validation_in_init_with_kwarg():
     assert Flow(name="test", edges=[e1, e2], validate=False)
 
 
-@pytest.mark.xfail(raises=ImportError, reason="viz extras not installed.")
+try:
+    import graphviz
+
+    graphviz.pipe("dot", "png", b"graph {a -- b}", quiet=True)
+    no_graphviz = False
+except Exception:
+    no_graphviz = True
+
+
+@pytest.mark.skipif(no_graphviz, reason="viz extras not installed.")
```

Comment:
This is way clearer (and faster!), but isn't it true that `xfail(raises=ImportError)` will properly `FAIL` if the error type is something different?

Reply:
That's a good point, I missed that kwarg. Note that you can have the python `graphviz` module installed without the cli `graphviz` library, so an `ImportError` isn't the only reason to skip these tests.